src/element/annotations.rs

```diff
@@ -0,0 +1,110 @@
+use crate::element::iterators::SymbolsIterator;
+use crate::Symbol;
+
+/// An ordered sequence of symbols that convey additional, application-specific information about
+/// their associated Ion value.
+/// ```
+/// use ion_rs::element::Annotations;
+/// let annotations: Annotations = ["foo", "bar", "baz"].into();
+/// for annotation in &annotations {
+///     assert_eq!(annotation.text().map(|s| s.len()), Some(3));
+/// }
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Annotations {
+    symbols: Vec<Symbol>,
+}
+
+impl Annotations {
+    // This is limited to crate visibility to allow us to change `Vec<_>` to something
+    // else (`thinvec`?) in the future.
+    pub(crate) fn new(symbols: Vec<Symbol>) -> Self {
+        Annotations { symbols }
+    }
+
+    // Constructs an Annotations object representing an empty symbol sequence
+    pub(crate) fn empty() -> Self {
+        Annotations {
+            symbols: Vec::new(),
+        }
+    }
+
+    /// Returns a [SymbolsIterator] that yields each of the [Symbol]s in this annotations sequence
+    /// in order.
+    pub fn iter(&self) -> SymbolsIterator {
+        self.into_iter()
+    }
+
+    /// Returns the number of annotations in this sequence.
+    /// ```
+    /// use ion_rs::element::Annotations;
+    /// let annotations: Annotations = ["foo", "bar", "baz"].into();
+    /// assert_eq!(annotations.len(), 3);
+    /// ```
+    pub fn len(&self) -> usize {
+        self.symbols.len()
+    }
+
+    /// Returns `true` if this sequence contains zero annotations. Otherwise, returns `false`.
+    /// ```
+    /// use ion_rs::element::Annotations;
+    /// let annotations: Annotations = ["foo", "bar", "baz"].into();
+    /// assert!(!annotations.is_empty());
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.symbols.len() == 0
+    }
+
+    /// Returns `true` if any symbol in this annotations sequence is equal to the provided text.
+    /// Otherwise, returns `false`.
+    /// ```
+    /// use ion_rs::element::Annotations;
+    /// let annotations: Annotations = ["foo", "bar", "baz"].into();
+    /// assert!(annotations.contains("foo"));
+    /// assert!(annotations.contains("bar"));
+    /// assert!(annotations.contains("baz"));
+    ///
+    /// assert!(!annotations.contains("quux"));
+    /// assert!(!annotations.contains("quuz"));
+    /// ```
+    pub fn contains<S: AsRef<str>>(&self, query: S) -> bool {
+        let query: &str = query.as_ref();
+        self.iter().any(|symbol| symbol.text() == Some(query))
+    }
```

Comment:
🗺️ While the implementations of `contains()` and `starts_with()` are not complex, they are needed commonly enough that providing convenience functions for them seems reasonable.